tomviz/PythonGeneratedDatasetReaction.cxx

```diff
@@ -328,6 +346,78 @@ void PythonGeneratedDatasetReaction::addDataset()
     shapeWidget->getShape(shape);
     this->dataSourceAdded(generator.createDataSource(shape));
   }
+  else if (this->Internals->scriptLabel == "Random Particles")
+  {
+    QDialog dialog;
+    dialog.setWindowTitle("Random Particles");
+    QVBoxLayout* layout = new QVBoxLayout; //overall layout
+    //Guide
+    QLabel *guide = new QLabel;
+    guide->setText("Generate many random 3D \"particles\" using the Fourier Noise method. You can increase the \"Internal Complexity\" of particles and their average \"Particle Size\". You can also specify the sparsity (percentage of non-zero voxels) of the generated dataset. Note: 512x512x512 may take a couple minutes to run.");
+    guide->setWordWrap(true);
+    layout->addWidget(guide);
+
+    //Shape Layout
+    ShapeWidget *shapeLayout = new ShapeWidget(&dialog);
+    shapeLayout->setSpinBoxValue(128,128,128);
+    shapeLayout->setSpinBoxMaximum(512,512,512);
```

Comment:
1024 takes a long time (~1hr). It is not reasonable to let the user wait this long. 512 seems reasonable for now.